### PR TITLE
OIDC_END_SESSION_ENDPOINT

### PR DESCRIPTION
**Pull request by gkostin1966**

### CODE REVIEW

File: app/config.rb [hunk 1]
```diff
@@ -43,11 +47,15 @@
...
+          redirect_uri: "#{AppConfig[:frontend_proxy_url]}auth/openid_connect/callback",
+          end_session_endpoint: oidc_end_session_endpoint
         }
       }
-   }
+    }
   ]
+
+  puts "OIDC settings were found; adding aspace-oauth plugin."
+  AppConfig[:plugins] << "aspace-oauth"
```
Comment (ssciolla) [inline app/config.rb:58]: Cool, yeah, I like having this in the `if` clause. I was seeing OmniAuth errors before when it wasn't enabled (not ones that blocked things, but still there).

File: app/env.example [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
+FRONTEND_PROXY_URL=http://localhost:8080
 # With deployments, use a secure unpredictable secret for DEFAULT_ADMIN_PASSWORD
 DEFAULT_ADMIN_PASSWORD=admin_pw
-
-# PUBLIC_URL should be your FQDN in deployments; this is used for the OIDC callback URL
-PUBLIC_URL=http://localhost:8080
 OIDC_ISSUER=
 OIDC_CLIENT_ID=
 OIDC_CLIENT_SECRET=
+OIDC_END_SESSION_ENDPOINT=
```
Comment (ssciolla) [inline app/env.example:10]: I'm not seeing a redirect included with the Shibboleth logout page, is that something you're working on in the plugin side? I think you were trying to use the frontend proxy URL for that.

**Review by ssciolla — APPROVED** [review]
Cool! I'm guessing there may be a little more work on the plugin side, but it's working for me, these changes look good.

File: app/env.example (hunk 2 above)
Comment (gkostin1966) [reply app/env.example:10]: The redirect works but you’ll need to include it in the `OIDC_END_SESSION_ENDPOINT=https://shibboleth.umich.edu/cgi-bin/logout?https://lib.umich.edu` … the proxy URL may not be the desired redirect e.g. like the library’s home page may be desired by the stakeholders.

Comment (ssciolla) [reply app/env.example:10]: Huh, I thought I tried it and it didn't work, but I tried again and seems fine. Must have had the URL wrong.

**Pull request merged**